src/main/java/org/opensearch/knn/index/engine/MethodComponent.java

```diff
@@ -328,19 +330,42 @@ public static Map<String, Object> getParameterMapWithDefaultsAdded(
         Map<String, Object> parametersWithDefaultsMap = new HashMap<>();
         Map<String, Object> userProvidedParametersMap = methodComponentContext.getParameters();
         Version indexCreationVersion = knnMethodConfigContext.getVersionCreated();
+        Mode mode = knnMethodConfigContext.getMode();
+        CompressionLevel compressionLevel = knnMethodConfigContext.getCompressionLevel();
```

Comment:
if mode and compression is not used and just encoder is provided then how we are changing the hyper parameters?

Reply:
this isOnDiskWithBinaryQuantization will be false ..and flow as it is like now

Reply:
that's why I was very specific to BinaryQuantization

Reply:
Ref: https://github.com/opensearch-project/k-NN/pull/2087/files#r1755277111

Reply:
Addressed